src/python/pants/option/custom_types.py

```diff
@@ -15,6 +15,13 @@
 from pants.util.strutil import ensure_text
 
 
+# A default value for a bool typed option that indicates it can act as a tri-state that defaults to
+# `None`; ie: the option is un-set.
+#
+# :API: public
+UnsetBool = object()
```

Comment:
this feels odd.. any reason to not just use an empty class here instead?

```
class UnsetBool(object):
  """...docstring..."""
```

this would allow for a proper docstring where `:API: public` makes more sense and `default is UnsetBool` would still work.